homeassistant/components/googleactions.py

```diff
@@ -0,0 +1,390 @@
+# pylint: disable=using-constant-test,unused-import,ungrouped-imports
+# if False:
+from homeassistant.core import HomeAssistant  # NOQA
+from aiohttp.web import Request, Response  # NOQA
+from typing import Dict, Tuple, Any  # NOQA
+from homeassistant.helpers.entity import Entity  # NOQA
+
+from homeassistant.helpers import config_validation as cv
+from homeassistant.components.http import HomeAssistantView
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    SERVICE_VOLUME_SET,
+    SERVICE_OPEN_COVER,
+    SERVICE_CLOSE_COVER,
+    SERVICE_SET_COVER_POSITION,
+    STATE_OFF,
+    HTTP_BAD_REQUEST,
+    HTTP_UNAUTHORIZED,
+    HTTP_MOVED_PERMANENTLY,
+    ATTR_SUPPORTED_FEATURES,
+    CONF_FRIENDLY_NAME, )
+from homeassistant.components.light import (ATTR_BRIGHTNESS,
+                                            SUPPORT_BRIGHTNESS)
+from homeassistant.components.media_player import ATTR_MEDIA_VOLUME_LEVEL
+
+_LOGGER = logging.getLogger(__name__)
+
+GOOGLE_ACTIONS_API_ENDPOINT = '/api/googleactions'
+
+DOMAIN = 'googleactions'
+DEPENDENCIES = ['http']
+
+ATTR_GOOGLE_ACTIONS = 'google_actions'
+ATTR_GOOGLE_ACTIONS_NAME = 'google_actions_name'
+
+CONF_EXPOSE_BY_DEFAULT = 'expose_by_default'
+CONF_EXPOSED_DOMAINS = 'exposed_domains'
+CONF_PROJECT_ID = 'project_id'
+CONF_ACCESS_TOKEN = 'access_token'
+CONF_CLIENT_ID = 'client_id'
+CONF_ALIASES = 'aliases'
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: {
+            vol.Required(CONF_PROJECT_ID): cv.string,
+            vol.Required(CONF_CLIENT_ID): cv.string,
+            vol.Required(CONF_ACCESS_TOKEN): cv.string,
+            vol.Optional(CONF_EXPOSE_BY_DEFAULT): cv.boolean,
+            vol.Optional(CONF_EXPOSED_DOMAINS): cv.ensure_list,
+        }
+    },
+    extra=vol.ALLOW_EXTRA)
+
+COMMAND_ONOFF = 'action.devices.commands.OnOff'
+COMMAND_BRIGHTNESS = 'action.devices.commands.BrightnessAbsolute'
+COMMAND_COLOR = 'action.devices.commands.ColorAbsolute'
+
+DEFAULT_EXPOSE_BY_DEFAULT = True
+DEFAULT_EXPOSED_DOMAINS = [
+    'switch', 'light', 'group', 'input_boolean', 'media_player', 'fan'
+]
+
+BASE_OAUTH_URL = 'https://oauth-redirect.googleusercontent.com'
+REDIRECT_TEMPLATE_URL = \
+    '{}/r/{}#access_token={}&token_type=bearer&state={}'
+
+
+def setup(hass: HomeAssistant, yaml_config: Dict[str, Any]):
+    """Activate Google Actions component."""
+    config = yaml_config.get(DOMAIN, {})
+
+    hass.http.register_view(GoogleActionsAuthView(hass, config))
+    hass.http.register_view(GoogleActionsView(hass, config))
+
+    return True
+
+
+class GoogleActionsAuthView(HomeAssistantView):
+    """Handle Google Actions auth requests."""
+
+    url = GOOGLE_ACTIONS_API_ENDPOINT + '/auth'
+    name = 'api:googleactions:auth'
+    requires_auth = False
+
+    def __init__(self, hass: HomeAssistant, cfg: Dict[str, Any]) -> None:
+        """Initialize instance of the view."""
+        super().__init__()
+        self.hass = hass
```

Comment:
You can get the `hass` instance from the request, you don't have to store it in the instance:

```
# inside get(…)
hass = request.app['hass']
```